pymatgen/command_line/chargemol_caller.py

```diff
@@ -170,34 +168,34 @@ def _execute_chargemol(self, **job_control_kwargs):
         """Internal function to run Chargemol.
 
         Args:
+            path (str): Path to the directory to run Chargemol in.
+                Default: None (current working directory).
             atomic_densities_path (str): Path to the atomic densities directory
             required by Chargemol. If None, Pymatgen assumes that this is
             defined in a "DDEC6_ATOMIC_DENSITIES_DIR" environment variable.
                 Default: None.
             job_control_kwargs: Keyword arguments for _write_jobscript_for_chargemol.
         """
-        with ScratchDir("."):
-            try:
-                os.symlink(self._chgcar_path, "./CHGCAR")
-                os.symlink(self._potcar_path, "./POTCAR")
-                os.symlink(self._aeccar0_path, "./AECCAR0")
-                os.symlink(self._aeccar2_path, "./AECCAR2")
-            except OSError as exc:
-                print(f"Error creating symbolic link: {exc}")
-
-            # write job_script file:
-            self._write_jobscript_for_chargemol(**job_control_kwargs)
-
-            # Run Chargemol
-            with subprocess.Popen(CHARGEMOL_EXE, stdout=subprocess.PIPE, stdin=subprocess.PIPE, close_fds=True) as rs:
-                _stdout, stderr = rs.communicate()
-            if rs.returncode != 0:
-                raise RuntimeError(
-                    f"{CHARGEMOL_EXE} exit code: {rs.returncode}, error message: {stderr!s}. "
-                    "Please check your Chargemol installation."
-                )
-
-            self._from_data_dir()
+        with ScratchDir(".") as temp_dir:
+            # with tempfile.TemporaryDirectory() as temp_dir:
+
+            os.symlink(self._chgcar_path, "CHGCAR")
+            os.symlink(self._potcar_path, "POTCAR")
+            os.symlink(self._aeccar0_path, "AECCAR0")
+            os.symlink(self._aeccar2_path, "AECCAR2")
+
+            lines = self._write_jobscript_for_chargemol(
+                chgcar_path=os.path.join(temp_dir, "CHGCAR"),
+                **job_control_kwargs,
+            )
+            with open(os.path.join(temp_dir, "job_control.txt"), mode="w") as file:
+                file.write(lines)
+
+            response = subprocess.run(CHARGEMOL_EXE, capture_output=True, check=False)
+            _stdout = response.stdout.decode("utf-8")
+            response.stderr.decode("utf-8")
```

Comment:
do these two lines serve a purpose? i don't think these raise when `response.returncode != 0`? i think we need a test for the previously raised `RuntimeError`

Reply:
maybe we can just use `check=True`? We don't even need to capture response/stdout here as it is not used?

Reply:
yes, let's use `check=True`. it's a different error type [`CalledProcessError`](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) which I think is better than `RuntimeError` anyway but we should still add a test for this behavior.